### ISSUE TITLE
variable dont change

### ISSUE THREAD

**Issue by proEngineering** [barrettford/Octoprint-Usb_keyboard#24]
Hello, people,

because I don't have a display, I wanted to use this plugin to drive my printer.

With keys 1 to 5 I wanted to assign a different value to my variable distance. Unfortunately, that doesn't work. Although the variable distance is stored in the profile, it assumes no other value.

How can I fix the error?

![grafik](https://user-images.githubusercontent.com/31779148/110656230-0acf6500-81c0-11eb-8019-d0284b7fdf93.png)
![grafik](https://user-images.githubusercontent.com/31779148/110656352-29356080-81c0-11eb-8ba3-46bbb9452f69.png)
![grafik](https://user-images.githubusercontent.com/31779148/110656602-613ca380-81c0-11eb-96b4-666fbe4b88d9.png)
![grafik](https://user-images.githubusercontent.com/31779148/110656789-91844200-81c0-11eb-9331-bb61f7992582.png)

**Comment by barrettford** [comment]
Can you attach a copy of your profile json?  It looks like you aren't listening/saving the new value anywhere, but I can't be certain because you've only included an image of part of your config.

**Comment by proEngineering** [comment]
your right! Now it works. But it looks very complicated.
![grafik](https://user-images.githubusercontent.com/31779148/110662510-deb6e280-81c5-11eb-9e57-c407dab662f8.png)
how does it work properly?

I tried so much combinations, but thats the only thing that works

**Comment by barrettford** [comment]
So, the variable system was designed to be used in a different way.  Normally you'd set one key on your keyboard to listen_vars when pressed and save_vars when released.  So, your setup would look like this:
<key A> pressed: listen_vars, released: save_vars
<key B> variables: distance 1

Press <key A>, Press <key B>, Release <key B>, Release <key A>. This would set your variable.  In this manner, variable listening/saving behave similar to modifier keys.

Does this make sense?

**Comment by proEngineering** [comment]
Yes, that makes sense. but would be great if that were described in more detail somewhere else. :D

Thank you very much!

**Comment by proEngineering** [comment]
Yea, i found these informations. But my english ist not so good. I didnt understand i should use two keys.
